### PR TITLE
Saga update runner

### PR DESCRIPTION
**Pull request by ramonsmits**
Implements #198

### CODE REVIEW

File: src/PerformanceTests/Common/Scenarios/SagaUpdate/SagaUpdateRunner.cs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+partial class SagaUpdateRunner
+    : BaseRunner
+{
+    public static bool Shutdown;
+    readonly ILog Log = LogManager.GetLogger(nameof(SagaUpdateRunner));
+
+    protected override async Task Start(ISession session)
+    {
+        var maxConcurrencyLevel = ConcurrencyLevelConverter.Convert(Permutation.ConcurrencyLevel);
+        var seedSize = maxConcurrencyLevel * 2;
```
Comment (dvdstelt) [inline src/PerformanceTests/Common/Scenarios/SagaUpdate/SagaUpdateRunner.cs]: I've seen this quite a number of times. Isn't this something you could do in BaseRunner or some helper class perhaps?

**Comment by dvdstelt** [comment]
Rest seems legit, a lot of the same stuff going on. Do it once, okay. Do it twice, good as well. Do it three times, you should probably refactor. How many times is code like this in the repo?
On the other hand, I don't think it needs to become more complex by adding yet another base class or something.

What do you think?

File: src/PerformanceTests/Common/Scenarios/SagaUpdate/SagaUpdateRunner.cs (hunk 1 above)
Comment (ramonsmits) [inline src/PerformanceTests/Common/Scenarios/SagaUpdate/SagaUpdateRunner.cs]: Have been considering to but the calculated concurrency value as a protected property in the base class. Not a real saver but now not each scenario needs to be responsible for knowing how to calculate it.

Same for the logic that drains the queue although I have been thinking about creating a behavior for that to very early in the pipeline.

File: src/PerformanceTests/Common/Scenarios/BaseRunner.cs [hunk 2]
```diff
@@ -302,4 +307,21 @@ static IEnumerable<int> IterateUntilFalse(Func<bool> condition)
         var i = 0;
         while (condition()) yield return i++;
     }
+
+    protected async Task DrainMessages()
```
Comment (dvdstelt) [inline src/PerformanceTests/Common/Scenarios/BaseRunner.cs]: I think this method is called in the `Close()` method of every runner. If this is the case, isn't it better to add it to `Stop()` method in the `BaseRunner` until someone actually does NOT need it and another approach can be taken?

Comment (ramonsmits) [inline src/PerformanceTests/Common/Scenarios/BaseRunner.cs]: @dvdstelt check the latest commit

**Pull request merged**